promise.js

```
@@ -0,0 +1,76 @@
+const request = require('request');
+
+let getDataOfRequest = url => new Promise((resolve, reject) => {
+    request({
+        url,
+        headers: {
+            'User-Agent': 'Batman'
+        }
+    }, (error, response, body) => {
+        if (error) return reject(error);
+        try {
+            resolve(JSON.parse(body));
+        } catch (e) {
+            reject(e);
```

if an error is not handled somehow inside `catch` the `catch` is useless.
errors can be just thrown from `new Promise()` the effect is the same like when they rejected